codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -417,11 +453,11 @@ public ChannelFuture writeGoAway(ChannelHandlerContext ctx, int lastStreamId, lo
     private ChannelFuture writeGoAway(ChannelHandlerContext ctx, Http2Exception cause) {
         Http2Connection connection = connection();
         if (connection.isGoAway()) {
-            return ctx.newSucceededFuture();
+            return ctx.newFailedFuture(connectionError(INTERNAL_ERROR,
```

Comment:
This check is a bit redundant and the public method should take care of the check.  I'm removing this for now.